main.py

```diff
@@ -127,10 +145,57 @@ def update_db(
     conn.commit()
     conn.close()
 
+# Calculate total cost and tokens for detailed report
+def calculate_cost_and_tokens(result):
+
+    total_embedding_cost = None
+    total_embedding_tokens = None
+    total_llm_cost = None
+    total_llm_tokens = None
+        
+    # Extract 'extraction_result' from the result
+    extraction_result = result.get("extraction_result", [])
+        
+    if not extraction_result:
+        return None, None, None, None
```

Comment:
Since we are initiating the variables as `None` let's respond with variables itself. I thought we might initiate it as `0` hence suggested this earlier

Reply:
Sure, I'll make the changes